frontend/apps/crates/components/src/module/_common/edit/entry/dom.rs

```diff
@@ -107,10 +108,19 @@ pub fn render_page_body<Mode, Step, RawData, Base, Main, Sidebar, Header, Footer
         })
     }));
 
-    state.page_body_switcher.load(sig.for_each(|dom| {
-        let body = dominator::body();
-        body.set_inner_html("");
-        dominator::append_dom(&body, dom);
+    state.page_body_switcher.load(sig.for_each(clone!(state => move |dom| {
+        {
+            // Discard the previous body and set the current handle to None.
+            // This forces dominator to release all references held by this handle.
+            let current_handle = state.dom_body_handle.replace(None);
```

Comment:
`replace()` is necessary because `discard()` moves the value which can't happen if we're holding a reference to it.